### DISCUSSION TITLE
Windows dev environment setup

### DISCUSSION THREAD

**Discussion by temabu** [ietf-tools/datatracker#7707] [Help / Questions]
hi, I have WSL and datatracker running on docker desktop.  I presume I have to register a user in the local DB but this involves an email step.  What are the steps required to configure celery in the local docker container to work?  thanks

**Comment by jennifer-richards** [comment]
Can you say more about your configuration? 

The instructions at https://github.com/ietf-tools/datatracker/blob/main/docker/README.md will give you a working dev environment, including celery.

**Reply by temabu** [reply]
I installed as per your link, but when registering a new account I get these error messages:  Where do I configure the mail server fqdn and port and authentication?

INFO: datatracker:48: Exception while trying to send email from 'ietf-secretariat-reply@ietf.org' to ['test@somedomain.com'] subject 'Confirm registration at datatracker.ietf.org'
INFO: datatracker:48: SMTP Exception: <class 'ConnectionRefusedError'> : [Errno 111] Connection refused
INFO: datatracker:48:      Traceback: ['  File "/workspace/ietf/utils/mail.py", line 111, in send_smtp\n    conn_code, conn_msg = server.connect(SMTP_ADDR[\'ip4\'], SMTP_ADDR[\'port\'])\n', '  File "/usr/local/lib/python3.9/smtplib.py", line 341, in connect\n    self.sock = self._get_socket(host, port, self.timeout)\n', '  File "/usr/local/lib/python3.9/smtplib.py", line 312, in _get_socket\n    return socket.create_connection((host, port), timeout,\n', '  File "/usr/local/lib/python3.9/socket.py", line 844, in create_connection\n    raise err\n', '  File "/usr/local/lib/python3.9/socket.py", line 832, in create_connection\n    sock.connect(sa)\n']
INFO: datatracker:48: Exception while trying to send email from 'django-project@ietf.org' to ['action@ietf.org'] subject 'Datatracker error while sending email'

**Reply by temabu** [reply]
on configuration:
- windows 11
- - WSL installed
- - VS Code installed
- datatracker containers running - all tests pass

I see everything else e.g. at :8000 and postgres login page for instance. Just email that doesnt appear to work.

**Reply by jennifer-richards** [reply]
You may not need to create your own account. The development database dump has all passwords replaced with the string `password`, so you can log in as any existing user on your dev system. This is useful if you want to work on parts of the system that require specific permissions.

Using the default dev configuration, messages that would be sent via email will be echoed to the Django dev server log, visible as the console on the "app" container. If you scroll back in that, you should see a copy of the mail that would have been sent, starting with
```
INFO: datatracker:48: Saved outgoing email from 'IETF Secretariat <ietf-secretariat-reply@ietf.org>' to <testuser@example.com> id <172202554659.675.3892110364329267178@195e8e9e7e8f> subject 'Confirm registration at datatracker.ietf.org as Message[1255800]'
---------- MESSAGE FOLLOWS ----------
b'MIME-Version: 1.0'
b'Content-Type: text/plain; charset="utf-8"'
b'Content-Transfer-Encoding: 7bit'
b'From: IETF Secretariat <ietf-secretariat-reply@ietf.org>'
b'To: <testuser@example.com>'
b'Subject: Confirm registration at datatracker.ietf.org'
```
Later, there will be a line with a URL starting `https://datatracker.ietf.org/accounts/create/confirm/...`. If you change `https://datatracker.ietf.org` to `http://localhost:8000`, then that URL will allow you to confirm the account creation.

If you want to connect to an actual SMTP service, you can set `EMAIL_HOST` and `EMAIL_PORT` in your `ietf/settings_local.py` to point to the server to which mail should be delivered. You'll also need to set `USING_DEBUG_EMAIL_SERVER=True`. If that service is local to your dev machine, instead of `localhost` you may need to use `host.docker.internal` to connect to host ports from within Docker.
